### PR TITLE
Feature/adding docker GitHub action

### PR DESCRIPTION
**Pull request by tokamak-git**
issue: https://github.com/cosmos/gaia/issues/916

This pr enables the creation and publishing of official gaia images to ghcr (github container registry)

 have added the details regarding the triggers for the action to the readme.

### CODE REVIEW

**Comment by faddat** [comment]
this will work, when merged

:/

ghcr is like that sadly

**Comment by tokamak-git** [comment]
closes #916

**Comment by tokamak-git** [comment]
> this will work, when merged
> 
> :/
> 
> ghcr is like that sadly

@faddat turns out it was a different how GitHub actions parses strings vs act the utility to run actions.

File: .github/workflows/docker-push.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+name: Docker - ghcr Build and Push
+
+on:
+  push:
+    schedule:
+      - cron: "0 10 * * *"
```
Comment (alexanderbez) [inline .github/workflows/docker-push.yml:6]: Why is there a need for a cron? Should pushes to release branches suffice?

File: Makefile [hunk 2]
```diff
@@ -197,7 +197,7 @@ endif
 .PHONY: run-tests $(TEST_TARGETS)
 
 docker-build-debug:
```
Comment (alexanderbez) [inline Makefile:199]: nit: undo. This dockerfile is solely for the e2e tests. It shouldn't be the image we push and use for operators on mainnet.

File: tests/e2e/genesis.go [hunk 3]
```diff
@@ -67,7 +67,10 @@ func addGenesisAccount(path, moniker, amountStr string, accAddr sdk.AccAddress)
 	}
 
 	if accs.Contains(accAddr) {
-		return fmt.Errorf("failed to add account to genesis state; account already exists: %s", accAddr)
+		return fmt.Errorf(
```
Comment (alexanderbez) [inline tests/e2e/genesis.go:70]: was this causing lint failure?

File: .github/workflows/docker-push.yml (hunk 1 above)
Comment (tokamak-git) [reply .github/workflows/docker-push.yml:6]: True, the pushes to release and tagging would suffice moving forward, but how do we trigger the previous images?
The solution  i came up with was using cron flag to trigger a job run that would pickup old tags and release branches.
At least that's how I hope it works.

Comment (tokamak-git) [reply .github/workflows/docker-push.yml:6]: we could remove it of course once legacy images are built

File: Makefile (hunk 2 above)
Comment (tokamak-git) [reply Makefile:199]: Wouldn't we want to run our tests in an environment as close to production as possible?
I can revert the change and create a dedicated Dockerfile. I have one somewhere. But what pitfalls do you see of us using the same image?

**Comment by tac0turtle** [comment]
Is help needed to get this over the finish line?

**Comment by tokamak-git** [comment]
incorrect merge branch, closing

**Comment by tokamak-git** [comment]
closing in favour of #1552

**Pull request closed without merging**